File: frontend/src/objects/merchant/apiTypes/UpdateProductRequest.ts
import type { ListingStatus } from '@/objects/shared/ids'
import type { ProductBundleGroup } from '@/objects/merchant/Product'

export interface UpdateProductRequest {
  name: string
  description: string
  imageUrl: string
  categoryName: string
  price: number
  remainingStock: number
  listingStatus: ListingStatus
  bundleGroups?: ProductBundleGroup[]
}

File: backend/src/merchant/objects/apiTypes/UpdateProductRequest.scala
package delivery.merchant.objects.apiTypes

import delivery.merchant.objects.{ProductBundleGroup, ProductInventoryMode}
import delivery.domain.ListingStatus

final case class UpdateProductRequest(
    name: String,
    description: String,
    imageUrl: String,
    categoryName: String,
    price: Double,
    remainingStock: Int,
    listingStatus: ListingStatus,
    inventoryMode: Option[ProductInventoryMode] = None,
    maxPerOrder: Option[Int] = None,
    bundleGroups: Option[List[ProductBundleGroup]] = None
)
